### PR TITLE
Partners page. 

### PR DESCRIPTION
**Pull request by benzenwen**
Adds a page with a section for Technology Partners (e.g. AWS, GCP, Azure, DO) and a section for System Integrator Partners.

### CODE REVIEW

File: .dir-locals.el [hunk 1]
```diff
@@ -0,0 +1,10 @@
+((js-mode . ((js-indent-level . 4)
```
Comment (sethvargo) [inline .dir-locals.el]: I don't think you meant to check this in

Comment (benzenwen) [inline .dir-locals.el]: Good catch.  Thought I had fixed that.  Didn't mean to start a religious war.

File: .dir-locals.el [hunk 2]
```diff
@@ -0,0 +1,9 @@
+((js-mode . ((js-indent-level . 2)
```
Comment (pearkes) [inline .dir-locals.el]: Don't think we need this file

**Comment by pearkes** [comment]
Doesn't look like the page uses source/images/HashiCorp-partners-stacked-black-greyscale-sticker.png, we can probs remove that. Otherwise LGTM

File: .dir-locals.el (hunk 2 above)
Comment (benzenwen) [inline .dir-locals.el]: Us emacs folks would appreciate having it there :)

Comment (pearkes) [inline .dir-locals.el]: Okey dokey

**Pull request merged**